bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java

```diff
@@ -30,67 +35,121 @@
+                .findFirst().orElseGet(() -> null);
+    }
 
     public HomekitTaggedItem(Item item, ItemRegistry itemRegistry) {
         this.item = item;
-        for (String tag : item.getTags()) {
 
-            if (item instanceof ColorItem) {
-                tag = "Colorful" + tag;
-            } else if (item instanceof DimmerItem) {
-                tag = "Dimmable" + tag;
+        try {
+            homekitAccessoryType = findAccessoryType(item);
+            homekitCharacteristicType = findCharacteristicType(item);
+            if (homekitAccessoryType != null && homekitCharacteristicType != null) {
+                throw new BadItemConfigurationException(
+                        "Items cannot be tagged as both a characteristic and an accessory type");
             }
+            List<GroupItem> matchingGroupItems = findMyAccessoryGroups(item, itemRegistry);
 
-            /*
-             * Is the item part of a tagged group AND does it have a matching CharacteristicType ?
-             * This matches items with tags that require a parent group like the "TargetTemperature" in
-             * thermostats
-             */
-            if (isMemberOfRootGroup(item, itemRegistry)) {
-                homekitCharacteristicType = HomekitCharacteristicType.valueOfTag(tag);
-            }
+            switch (matchingGroupItems.size()) {
+                case 0: // Does not belong to a accessory group
+                    if (homekitCharacteristicType != null) {
+                        throw new BadItemConfigurationException(
+                                "Item is tagged as a characteristic, but does not belong to a root accessory group");
+                    }
 
-            /*
-             * If its not a characteristic type for a group item, see if we have a matching device type.
-             */
-            if (homekitCharacteristicType == null) {
-                homekitDeviceType = HomekitDeviceType.valueOfTag(tag);
-            }
+                    parentGroupItem = null;
+                    break;
+                case 1: // Belongs to exactly one accessory group
+                    if (item instanceof GroupItem) {
+                        throw new BadItemConfigurationException("Nested Accessory Groups are not supported");
+                    }
 
-            if (homekitDeviceType != null || homekitCharacteristicType != null) {
-                break;
+                    parentGroupItem = matchingGroupItems.get(0);
+                    break;
+                default: // Belongs to more than one accessory group
+                    throw new BadItemConfigurationException(
+                            "Item belongs to multiple Groups which are tagged as Homekit devices.");
             }
+
+        } catch (BadItemConfigurationException e) {
+            logger.error("Item {} was misconfigured: {}. Excluding item from homekit.", item.getName(), e.getMessage());
```

Comment:
I think a lot of the error logging by this addon should be done using warn instead. 

See [logging guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging):

> `error` logging is not allowed in extensions and is purely reserved to the framework. The only exception would be if something is going really, really wrong in your extension, and there is a possibility that the stability of the framework could be affected.